pebble.go

```diff
@@ -136,6 +136,29 @@ func (db *PebbleDB) DB() *pebble.DB {
 	return db.db
 }
 
+func (db *PebbleDB) Compact(start, end []byte) error {
+	// Currently nil,nil is an invalid range in Pebble.
+	// This was taken from https://github.com/cockroachdb/pebble/issues/1474
+	// In case the start and end keys are the same
+	// pebbleDB will throw an error that it cannot compact.
+	if start != nil && end != nil {
+		return db.db.Compact(start, end, true)
+	}
+
+	iter := db.db.NewIter(nil)
+
+	if start == nil && iter.First() {
+		start = append(start, iter.Key()...)
+	}
+	if end == nil && iter.Last() {
+		end = append(end, iter.Key()...)
+	}
+	if err := iter.Close(); err != nil {
```

Comment:
Can we close the iterator in a defer? In case the calls to `iter.XXX` in the lines above unwind the stack...

Reply:
Sure, will do.

Reply:
Done in https://github.com/cometbft/cometbft-db/pull/111/commits/ea11e09cda6b2d32b72e20416d8c39bf85c33abd